### PR TITLE
Fix 5032

### PR DESCRIPTION
**Pull request by wochinge**
**Proposed changes**:
- fix https://github.com/RasaHQ/rasa/issues/5032
- Fix wrong calculation of additional conversation events when saving the conversation.
This led to conversation events not being saved (we always calculated number of events from the start, but the retrieved tracker only contains the events since the last session start)
- Fix wrong order of conversation events when pushing events to conversations via
``POST /conversations/<conversation_id>/tracker/events``.

**Status (please check what you already did)**:
- [x] added some tests for the functionality
- [ ] updated the documentation
- [x] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/master/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

**Comment by wochinge** [comment]
We have to release a patch a soon as this is merged 🔥

**Comment by tmbo** [comment]
Sounds good 👌

File: rasa/core/tracker_store.py [hunk 1]
```diff
@@ -765,6 +744,33 @@ def retrieve(self, sender_id: Text) -> Optional[DialogueStateTracker]:
                 )
                 return None
 
+    def _event_query(self, session: "Session", sender_id: Text) -> "Query":
```
Comment (ricwo) [inline rasa/core/tracker_store.py]: it would be great if you could add a comment

Comment (ricwo) [inline rasa/core/tracker_store.py]: it would be great if you could add a comment

File: rasa/server.py [hunk 2]
```diff
@@ -466,7 +466,30 @@ def create_app(
...
 
+        verbosity = event_verbosity_parameter(request, EventVerbosity.AFTER_RESTART)
+
+        try:
+            async with app.agent.lock_store.lock(conversation_id):
+                tracker = await get_tracker(
+                    app.agent.create_processor(), conversation_id
+                )
+
+                events = _get_events_from_request_body(request)
```
Comment (ricwo) [inline rasa/server.py]: this can / should happen before the tracker is fetched, for instance right away in line 468

Comment (wochinge) [reply rasa/server.py]: I added a comment to clarify why I'm doing it like this. 
The problem is that the events get assigned new timestamps when they are parsed from the request payload. However, also the tracker initialization might lead to events such as the session started event which should be **before** the appended events. Does that make sense?

**Comment by wochinge** [comment]
@ricwo If you are working today / tomorrow it would be great if you can just merge it / edit + merge it when you have time. Thanks :-)

File: rasa/server.py (hunk 2 above)
Comment (ricwo) [reply rasa/server.py]: makes sense, thanks for adding that comment

**Pull request merged**